services/brig/test/unit/Test/Brig/Calling/Internal.hs

```diff
@@ -0,0 +1,48 @@
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
+
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2020 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module Test.Brig.Calling.Internal where
+
+import Brig.Calling.Internal
+import Data.Misc (mkHttpsUrl)
+import Imports
+import Test.Tasty
+import Test.Tasty.HUnit
+import URI.ByteString.QQ as URI
+import Wire.API.Call.Config (sftServer)
+import Wire.Network.DNS.SRV (SrvTarget (SrvTarget))
+
+tests :: TestTree
+tests =
+  testGroup "Calling.API" $
+    [ testGroup "sftServerFromSrvTarget" $
+        [ testCase "when srvTarget ends with a dot" $ do
+            let Right expectedServer = sftServer <$> mkHttpsUrl [URI.uri|https://sft1.env.example.com:9364|]
+            assertEqual
+              "the dot should be stripped from sft server"
```

Comment:
The dot stripping, and converting of srvTarget to HttpsUrl seems more general than only useful in brig calling; i.e. I think we can also make use of that logic for federation.

Do you think this test and the related function could be moved from Brig calling to dns-util ?

Reply:
I would say let's move it when we need to, I think the `HttpsUrl` type could use some refactoring to make sure there is at least one way to create it without running into the `Either`.

Reply:
> I think the `HttpsUrl` type could use some refactoring to make sure there is at least one way to create it without running into the `Either`.

I created another smart constructor.

Reply:
> Otherwise, we can also leave a FUTUREWORK here.

I did this.